merkletree/proof.go

```diff
@@ -80,38 +93,41 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
-	return bytes.Equal(ap.Leaf.Value, value) &&
-		ap.Leaf.Commitment.Verify(key, value)
-}
-
-// Verify recomputes the tree's root node from the authentication path,
+// Verify first compares the lookup index with the leaf index.
+// It expects the lookup index and the leaf index match in the
+// first l bits with l is the Level of the proof node if ap is
+// a proof of absence. It also verifies the value and
+// the commitment (in case of the proof of inclusion).
+// Finally, it recomputes the tree's root node from ap,
 // and compares it to treeHash, which is taken from a STR.
-// Specifically, treeHash has to come from the STR whose tree returns
-// the authentication path.
+// Specifically, treeHash has to come from the STR whose tree returns ap.
+//
 // This should be called after the VRF index is verified successfully.
-func (ap *AuthenticationPath) Verify(key, value, treeHash []byte) bool {
-	if ap.ProofType() == ProofOfAbsence { // proof of absence
+func (ap *AuthenticationPath) Verify(key, value, treeHash []byte) error {
+	if ap.ProofType() == ProofOfAbsence {
 		// Check if i and j match in the first l bits
 		indexBits := utils.ToBits(ap.Leaf.Index)
 		lookupIndexBits := utils.ToBits(ap.LookupIndex)
 		for i := 0; i < int(ap.Leaf.Level); i++ {
 			if indexBits[i] != lookupIndexBits[i] {
-				return false
+				return ErrIndicesMismatch
 			}
 		}
-	} else { // proof of inclusion
+		if ap.Leaf.Value != nil {
+			return ErrBindingsDiffer
```

Comment:
Ok, thank you, I forgot we did that.  Is it worth putting a comment here for people like me who don't remember so well?